riscv-elf.md

```diff
@@ -420,6 +422,12 @@ arguments are either `0` (`false`) or `1` (`true`).
 
 A null pointer (for all types) has the value zero.
 
+`__fp16` type is treated as storage specifiers and promoted to single-precision
+`float` or double-precision `dobule` before any arithmetic operations or
```

Comment:
I would suggest do not mention `double` here, since the only place `__fp16` will promote to `double` is the variadic arguments, and `formatting operations` is not formal term, so keep single-precision should be enough.
```
`__fp16` type is treated as storage specifiers and promoted to single-precision
`float`  before any arithmetic operations.
```